src/EnergyPlus/ScheduleManager.cc

```diff
@@ -1641,354 +1636,229 @@ namespace ScheduleManager {
+        // While walking through the file,
+        // filename,colNumber  is  corresponding to ColumnarData.index
+        // correct row - dont check now
+        // columnarData[<column> index].vals[<row>] = token being read
+
+        // for item in setOfFilenames fill in appropriate  cells of columnar data
+        for (const std::string &fileNameItem : setOfFilenames) {
+            // go through all in allIdfSchedData, filter which sched belong to this file
+            // get their column map
+            std::map<int, int> colNumToColDataIndex;
+            std::vector<int> vectorOfCSVColNumbers;
+            for (const schedInputIdfPreprocessObject &item : state.dataScheduleMgr->allIdfSchedData) {
+                if (item.fileName == fileNameItem) {
+                    colNumToColDataIndex[item.columnOfInterest] = item.columnarDataIndex;
+                    vectorOfCSVColNumbers.push_back(item.columnOfInterest);
+                }
             }
 
-            if (lAlphaBlanks(4) || UtilityRoutines::SameString(Alphas(4), "comma")) {
-                ColumnSep = CharComma;
-                Alphas(4) = "comma";
-            } else if (UtilityRoutines::SameString(Alphas(4), "semicolon")) {
-                ColumnSep = CharSemicolon;
-            } else if (UtilityRoutines::SameString(Alphas(4), "tab")) {
-                ColumnSep = CharTab;
-            } else if (UtilityRoutines::SameString(Alphas(4), "space")) {
-                ColumnSep = CharSpace;
-            } else {
-                ShowSevereError(state,
-                                RoutineName + CurrentModuleObject + "=\"" + Alphas(1) + "\", " + cAlphaFields(4) + " illegal value=\"" + Alphas(4) +
-                                    "\".");
-                ShowContinueError(state, "..must be Comma, Semicolon, Tab, or Space.");
-                ErrorsFound = true;
-                continue;
-            }
+            // find if there is a many to one mapping, throw fatal error if there is an error
+            std::sort(vectorOfCSVColNumbers.begin(), vectorOfCSVColNumbers.end()); // sorting to use std::adjacent_find
+            const auto duplicate =
+                std::adjacent_find(vectorOfCSVColNumbers.begin(), vectorOfCSVColNumbers.end()); // what if there are multiple last elements?
+            if (duplicate !=
+                vectorOfCSVColNumbers
+                    .end()) { // TODO : what if there are multiple last elements? - remove last element and check if duplicate is still the same
+                ShowFatalError(state,
+                               format("Multiple schedules pointing to the same column number : {} in file {}. Please modify the IDF file and make "
+                                      "sure multiple schedules do not point to the same column in the file.",
+                                      *duplicate,
+                                      fileNameItem));
+            }
```

Comment:
Having multiple schedules pointing to the same column in a CSV creates a many-to-one mapping, which breaks the code here. So I set up a check and a fatal error if this was happening. The only reason I see someone wanting this would be if they wanted to stack multiple schedules in one CSV column, using the "rows to skip" field.

Reply:
Would it be difficult to let two schedule names share the same data column? I could see doing that intentionally as a base case and then changing the column for one schedule but not another in another run. It's not a huge deal, the user could simply duplicate a column as needed.

Reply:
Thanks for the review!
I spent a bit of time thinking about how to implement this, and also debating whether the extra code complexity was worth the feature. I will add it to the TODO list :)